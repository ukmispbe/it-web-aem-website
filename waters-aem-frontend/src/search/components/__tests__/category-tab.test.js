import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import screenSizes from '../../../scripts/screenSizes';
import CategoryTab from '../category-tab';

describe('Feature: CategoryTab React Component', () => {
    const props = {
        name: 'Tab1',
        translation: 'Tab Uno',
        count: 0,
        isActive: false,
        index: 0,
        onClick: jest.fn(() => {})
    };

    describe('Scenario: Rendering', () => {
        describe('When the screen size is tablet or mobile', () => {
            it('Then it should not render anything and match snapshot', () => {
                screenSizes.isTabletAndUnder = jest.fn(() => true);
                const json = renderer.create(<CategoryTab {...props}/>);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When the screen size is desktop and tab is inactive', () => {
            it('Then is should render the component and match snapshot', () => {
                screenSizes.isTabletAndUnder = jest.fn(() => false);
                const json = renderer.create(<CategoryTab {...props}/>);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When the screen size is desktop and tab is active', () => {
            it('Then is should render the component and match snapshot', () => {
                props.isActive = true;
                screenSizes.isTabletAndUnder = jest.fn(() => false);
                const json = renderer.create(<CategoryTab {...props}/>);
                expect(json).toMatchSnapshot();
            });
        });
    });

    describe('Scenario: User Interaction', () => {
        describe('When the tab is clicked', () => {
            it('It should execute the onClick prop', () => {
                screenSizes.isTabletAndUnder = jest.fn(() => false);
                const onClickSpy = spyOn(props, 'onClick');
                const wrapper = shallow(<CategoryTab {...props} />);
                wrapper.find('.cmp-search__categories-tabs--tab').simulate('click');

                expect(onClickSpy).toHaveBeenCalled();

                onClickSpy.mockRestore();
            });
        });
    });
});