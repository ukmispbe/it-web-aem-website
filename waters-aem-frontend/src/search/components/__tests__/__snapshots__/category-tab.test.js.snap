// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: CategoryTab React Component Scenario: Rendering When the screen size is desktop and tab is active Then is should render the component and match snapshot 1`] = `
<div
  className="cmp-search__categories-tabs--tab active"
  onClick={[Function]}
>
  <span
    className="cmp-search__categories-tabs--tab-name"
  >
    Tab1
  </span>
</div>
`;

exports[`Feature: CategoryTab React Component Scenario: Rendering When the screen size is desktop and tab is inactive Then is should render the component and match snapshot 1`] = `
<div
  className="cmp-search__categories-tabs--tab"
  onClick={[Function]}
>
  <span
    className="cmp-search__categories-tabs--tab-name"
  >
    Tab1
  </span>
</div>
`;

exports[`Feature: CategoryTab React Component Scenario: Rendering When the screen size is tablet or mobile Then it should not render anything and match snapshot 1`] = `null`;
